setup.sh

```diff
@@ -0,0 +1,74 @@
+#! /usr/bin/env bash
+
+# Bash strict mode
+set -euo pipefail
+
+echo "PORT=8083" > .env
+echo "EXTRA_PORT='8082'" >> .env
+echo "WEBPACK_DEV_URL=http://0.0.0.0:8082" >> .env
+
+echo "##########################"
+echo "#                        #"
+echo "#         Github         #"
+echo "#                        #"
+echo "##########################"
+echo "You need to register a new OAuth application:"
+echo "https://github.com/settings/applications/new"
+echo
+echo "You can put those parameters:"
+echo "* Application name: BSI development"
+echo "* Homepage URL: 0.0.0.0:8083"
+echo "* Application description: This a development OAuth application for BSI"
+echo "* Authorization callback URL: http://0.0.0.0:8083/auth/verify"
+echo
+
+echo "Press any key to continue"
+read
+
+echo
+echo -n "Enter client id: "
+read client_id
+echo -n "Enter client secret: "
+read client_secret
+echo -n "Enter Github Username: "
+read username
+echo -n "Enter Github Password: "
+read -s password
+echo -en "\nEnter Github 2FA: "
+read fa
+
+user="$username:$password"
+content_type="Content-Type: application/json"
+otp="X-GitHub-OTP: $fa"
+data="{
+  \"client_id\": \"${client_id}\",
+  \"client_secret\": \"${client_secret}\",
+  \"note\": \"Build.s.io poller token\"
+}"
+
+token=$(curl -X POST \
+     https://api.github.com/authorizations \
+     -u "$user" \
+     -H "$content_type" \
+     -H "$otp" \
+     -d "$data" | jq '.token' -r)
```

Comment:
While `jq` is certainly a useful tool, it isn't installed by default.

Reply:
we are going to run this script in our docker environment which has this package.